synapse/handlers/room_summary.py

```diff
@@ -803,8 +803,10 @@ async def _summarize_remote_room_hierarchy(
             if "room_id" in c and isinstance(c["room_id"], str)
         }
 
+        room_response = dict(room_response)
+        children_state = room_response.pop("children_state", ())
```

Comment:
Would it make more sense to make this a separate return value from `get_room_hierarchy` so that we do no modifications in here (and no copies)?

Reply:
Fewer extra copies sounds good!